addons/stock/models/stock_location.py

```diff
@@ -206,19 +206,18 @@ def _unlink_except_master_data(self):
             raise ValidationError(_('The %s location is required by the Inventory app and cannot be deleted, but you can archive it.', inter_company_location.name))
 
     def _search_is_empty(self, operator, value):
-        if operator not in ('=', '!=') or not isinstance(value, bool):
-            raise NotImplementedError(_(
-                "The search does not support the %(operator)s operator or %(value)s value.",
-                operator=operator,
-                value=value,
-            ))
-        groups = self.env['stock.quant']._read_group([
-            ('location_id.usage', 'in', ['internal', 'transit'])],
-            ['location_id'], ['quantity:sum'])
-        location_ids = {loc.id for loc, quantity in groups if quantity >= 0}
-        if value and operator == '=' or not value and operator == '!=':
-            return [('id', 'not in', list(location_ids))]
-        return [('id', 'in', list(location_ids))]
+        if operator != 'in':
+            raise NotImplementedError
+        location_ids = [
+            location_id
+            for location_id, _ in self.env['stock.quant']._read_group(
+                [('location_id.usage', 'in', ['internal', 'transit'])],
+                ['location_id'],
+                ['quantity:sum'],
```

Comment:
You actually don't need to ask any aggregates. having works fine even on stuff that are not in `aggregates`